workos/sso.py

```diff
@@ -121,69 +76,83 @@ def get_authorization_url(
             "response_type": RESPONSE_TYPE_CODE,
         }
 
-        if (
-            domain is None
-            and provider is None
-            and connection is None
-            and organization is None
-        ):
+        if connection_id is None and organization_id is None and provider is None:
             raise ValueError(
-                "Incomplete arguments. Need to specify either a 'connection', 'organization', 'domain', or 'provider'"
+                "Incomplete arguments. Need to specify either a 'connection', 'organization', or 'provider'"
             )
         if provider is not None:
-            if not isinstance(provider, SsoProviderType):
-                raise ValueError("'provider' must be of type SsoProviderType")
-
-            params["provider"] = provider.value
-        if domain is not None:
-            warn(
-                "The 'domain' parameter for 'get_authorization_url' is deprecated. Please use 'organization' instead.",
-                DeprecationWarning,
-            )
-            params["domain"] = domain
+            params["provider"] = provider
         if domain_hint is not None:
             params["domain_hint"] = domain_hint
         if login_hint is not None:
             params["login_hint"] = login_hint
-        if connection is not None:
-            params["connection"] = connection
-        if organization is not None:
-            params["organization"] = organization
+        if connection_id is not None:
+            params["connection"] = connection_id
+        if organization_id is not None:
+            params["organization"] = organization_id
 
         if state is not None:
             params["state"] = state
 
-        if redirect_uri is None:
-            raise ValueError("Incomplete arguments. Need to specify a 'redirect_uri'.")
+        fake_http_client = FakeHTTPClient(
+            base_url=self._http_client.base_url,
+            version=self._http_client.version,
+            timeout=self._http_client.timeout,
+        )
 
-        prepared_request = Request(
-            "GET",
-            self.request_helper.generate_api_url(AUTHORIZATION_PATH),
-            params=params,
-        ).prepare()
+        request_url = fake_http_client.build_request_url(
+            url=AUTHORIZATION_PATH, method=REQUEST_METHOD_GET, params=params
+        )
 
-        return prepared_request.url
+        fake_http_client.close()
```

Comment:
... this is the idiomatic way to construct a URL? That's truly Slytherin

Reply:
[Is `urlencode`](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlencode) out of favor?